src/common/range_spec.h

```diff
@@ -24,36 +24,36 @@
 
 #include "status.h"
 
-struct CommonRangeLexSpec {
+struct RangeLexSpec {
   std::string min, max;
   bool minex = false, maxex = false; /* are min or max exclusive */
   bool max_infinite = false;         /* are max infinite */
   int64_t offset = -1, count = -1;
   bool removed = false, reversed = false;
-  explicit CommonRangeLexSpec() = default;
+  explicit RangeLexSpec() = default;
 };
 
-Status ParseRangeLexSpec(const std::string &min, const std::string &max, CommonRangeLexSpec *spec);
+Status ParseRangeLexSpec(const std::string &min, const std::string &max, RangeLexSpec *spec);
 
-struct CommonRangeRankSpec {
+struct RangeRankSpec {
   int start, stop;
```

Comment:
Why start and stop doesn't have default value?

Reply:
It's somehow always set. But we can provide one?

If you take a closer look, `min` and `max` in the other two specs don't have default value also.

Reply:
Seems that you set default min max for `RangeScoreSpec` in previous patch.

`RangeLexSpec`'s min max is `std::string`, they have default construct value "empty string".

Not setting is ok, but I wonder if start/stop be an undefined value here.

Reply:
We'd better to give default values, otherwise the value is undetermined.

Reply:
OK. Let me set it to 0 and -1, correspondingly?

Reply:
Updated.